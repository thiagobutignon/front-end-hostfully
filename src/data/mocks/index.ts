export * from './http-client-spy'
