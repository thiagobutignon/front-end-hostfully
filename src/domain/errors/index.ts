export * from './email-in-use-error'
export * from './invalid-credentials'
export * from './not-found-error'
export * from './unexpected-erros'
