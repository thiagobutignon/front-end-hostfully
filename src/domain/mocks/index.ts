export * from './booking.mock'
export * from './guest.mock'
export * from './location.mock'
export * from './property.mock'
