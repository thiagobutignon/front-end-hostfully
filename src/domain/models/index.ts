export * from './booking.model'
export * from './guest.model'
export * from './location.model'
export * from './property.model'
