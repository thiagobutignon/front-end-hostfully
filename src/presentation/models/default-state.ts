export type DefaultState = {
  isLoading: boolean
  mainError: string
}