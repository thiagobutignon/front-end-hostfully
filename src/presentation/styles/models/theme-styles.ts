export type ThemeStyles = {
  semanticTokens: ThemeStyles.SemanticTokens
  styles: ThemeStyles.Styles
}
// eslint-disable-next-line @typescript-eslint/no-redeclare
export namespace ThemeStyles {
  export type ColorMode = {
    default: string
    _dark: string
  }

  export type Styles = {
    global: {
      body: {
        bgColor: string
        color: string
      }
    }
  }

  export type SemanticTokens = {
    colors: {
      background: ColorMode
      text: ColorMode
      detail: ColorMode
      buttonHud: ColorMode
      successButton: ColorMode
      error: ColorMode
      delete: ColorMode
      reverseBackground: ColorMode
      reverseText: ColorMode
      iconColor: ColorMode
      reverseIconColor: ColorMode
      borderColor: ColorMode
      cardColor: ColorMode
    }
  }
}
