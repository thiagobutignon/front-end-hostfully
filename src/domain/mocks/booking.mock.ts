import { guestModelMock, propertyModelMock } from '@/domain/mocks'

import { Booking } from '@/domain/models'
import { faker } from '@faker-js/faker'

export const bookingModelMock = (): Booking.Model => ({
  id: faker.string.alpha(),
  propertyId: faker.string.alphanumeric(),
  totalPrice: faker.number.float().toString(),
  startDate: faker.date.recent(),
  endDate: faker.date.future(),
  hostEmail: faker.internet.email(),
  guests: guestModelMock()
})

export const bookingParamsMock = (): Booking.Params => ({
  guestEmail: faker.internet.email(),
  propertyId: faker.string.alphanumeric(),
  guests: guestModelMock(),
  startDate: faker.date.recent(),
  endDate: faker.date.future(),
  createdAt: faker.date.recent(),
  property: propertyModelMock()
})

export const bookindResultMock = (): Booking.Result => ({
  booking: Array.from({ length: 10 }, () => bookingModelMock())
})
