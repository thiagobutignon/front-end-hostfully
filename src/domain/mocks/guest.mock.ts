import { Guest } from '@/domain/models'
import { faker } from '@faker-js/faker'

export const guestModelMock = (): Guest.Model => ({
  numberOfGuests: faker.number.int(),
  guests: Array.from({ length: 10 }, () => guestInfoMock())
})

const guestInfoMock = (): Guest.Info => ({
  name: faker.person.fullName(),
  email: faker.internet.email()
})
