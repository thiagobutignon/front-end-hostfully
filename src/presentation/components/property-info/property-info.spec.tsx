import { render, screen } from '@testing-library/react'

import PropertyInfoComponent from '@/presentation/components/property-info/property-info'
import { PropertyModel } from '@/domain/models'
import { propertyModelMock } from '@/domain/mocks'

describe('PropertyInfoComponent', () => {
  let selectedProperty: PropertyModel
  beforeEach(() => {
    selectedProperty = propertyModelMock()
  })
  it('should display property details correctly', () => {
    render(
      <PropertyInfoComponent selectedProperty={selectedProperty}>
        <div>children</div>
      </PropertyInfoComponent>
    )

    expect(screen.getByText(selectedProperty.name)).toBeInTheDocument()
    expect(
      screen.getByText(
        `Guests: ${selectedProperty.maxGuests} | Bedrooms: ${selectedProperty.bedrooms} | Beds: ${selectedProperty.beds}`
      )
    ).toBeInTheDocument()
    expect(screen.getByText(selectedProperty.description)).toBeInTheDocument()
    expect(
      screen.getByText(`$${selectedProperty.pricePerNight}/night`)
    ).toBeInTheDocument()
    expect(screen.getByRole('img')).toBeInTheDocument()
  })

  it.skip('matches snapshot', () => {
    const { asFragment } = render(
      <PropertyInfoComponent selectedProperty={selectedProperty}>
        <div>children</div>
      </PropertyInfoComponent>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
