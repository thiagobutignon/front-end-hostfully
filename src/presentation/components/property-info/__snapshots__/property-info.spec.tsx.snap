// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInfoComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-vjwr8m"
  >
    <img
      class="chakra-image css-afi9h6"
      src="https://loremflickr.com/640/480/house?lock=6096378224705536"
    />
    <h1
      class="chakra-heading css-1y1i1kg"
    >
      Pat Bashirian
    </h1>
    <p
      class="chakra-text css-1tg6nva"
    >
      Guests: 2111450424279048 | Bedrooms: 4512976035905536 | Beds: 6616751263449088
    </p>
    <p
      class="chakra-text css-1j9oeic"
    >
      Aedificium supra caelum amet bestia paulatim tolero aliquam arguo.
    </p>
    <div
      class="css-1gvcg4y"
    >
      <p
        class="chakra-text css-e5hj7u"
      >
        $0.6472861601505429/night
      </p>
      <button
        class="chakra-button css-16bvcfa"
        type="button"
      >
        Book Now
      </button>
    </div>
  </div>
</DocumentFragment>
`;
